sdk/inc/azure/core/az_json.h

```diff
@@ -390,6 +390,17 @@ az_json_writer_get_bytes_used_in_destination(az_json_writer const* json_writer)
 /**
  * @brief Appends the UTF-8 text value (as a JSON string) into the buffer.
  *
+ * @note For details on possibly receiving #AZ_ERROR_NOT_ENOUGH_SPACE while appending a string for
+ * which there is theoretically space, please see this github issue here for an explanation.
+ *
+ * https://github.com/Azure/azure-sdk-for-c/issues/1030
+ *
+ * The high level reasoning copied from the issue is below.
+ *
+ * "The JSON writer pessimistically requires at least 64-bytes of space when writing any string
+ * larger than 10 characters because it tries to write in 64 byte chunks (10 character * 6 if all
+ * need to be escaped into the unicode form)."
+ *
```

Comment:
Good idea!

However, rather than referencing a GitHub issue in the doc comments, let's just mention the reason in-line.

Something like what you summarized is good (we can leave out the issue). I would add:
> The JSON writer requires at least 64-bytes of slack within the output buffer, above the theoretical minimal space needed.

I can help wordsmith it some more in a bit :)

Question: `@remarks` vs `@note`, which renders better in this case?

Reply:
Also, this pessimistic approach and a bit of slack space isn't just need for strings, the same applies to any thing that's variable, including numbers and property names. Once we settle on the text, we should add such a note to those APIs too.

So, even calls to `append_int32(writer, 1)` would fail, if the buffer doesn't have enough slack, need at least 11 bytes even though 1 could, theoretically, be enough. For simplicity to users, just providing at least 64-bytes of padding above the "expected" size works across the board. The caller needs to be able to handle "not enough size" regardless, since JSON output is dynamic in nature, based on changing scenarios.

https://github.com/Azure/azure-sdk-for-c/blob/0ade924e2832ad187fc62d6a37691651a0334f0e/sdk/src/azure/core/az_json_writer.c#L861

Reply:
I've always used `@note` so usually stick with it. Updated the wordage let me know how that looks to you. I put it in other locations where it might be relevant as well.

Reply:
Difference between `@note` and `@remark` in doxygen rendering. It has a yellow bar to highlight it some more, when it is a `@note`:
![image](https://user-images.githubusercontent.com/6527137/140428930-d0825ced-f29f-420a-8298-caeb910a17f2.png)